apps/frontend/src/app/OrderList/OrderList.tsx

```diff
@@ -0,0 +1,37 @@
+import { CardActionArea, styled } from '@mui/material';
+import { OrderModel } from '@pasnik/api/data-transfer';
+import { useCallback } from 'react';
+import { useHistory } from 'react-router-dom';
+import OrderElement from '../OrderElement';
+
+export default function OrderList(props: { orders: OrderModel[] }) {
+  const history = useHistory();
+
+  const StyledCardActionArea = styled(CardActionArea)(() => ({
+    margin: 8,
+    padding: 8,
+    border: '1px solid black',
+    borderRadius: 10,
+    boxShadow: 'inset 0 7px 30px -10px rgba(150,170,180,0.5)',
+  }));
+
+  const routeChange = useCallback(
+    (id: string) => {
+      history.push(`/order/${id}`);
+    },
+    [history]
+  );
+
+  return (
+    <>
+      {props.orders.map((order) => (
+        <StyledCardActionArea
+          className="raise"
+          onClick={() => routeChange(order.id)}
```

Comment:
please avoid inline function in props as it will create new function every time component rerender causing the child to be rerendered as well

Reply:
I tried to do it by handler and I had a problem with passing prop 'order.id'. Found smth in the internet but still doesnt work. Tried also with .bind(this,order.id).  Alsoo thought about passing order.id as props to component and handling stuff there but its mui component so i guess thats dead end.
![image](https://user-images.githubusercontent.com/64810185/134640291-b36ba172-c02a-41cd-890d-eb6c2ea780a6.png)

Reply:
so you want to pass a prop to callback function but onClick also has a prop passing so you need to have a closure function,

const closureFn = (someProp) => (mouseClickEvent) => { //do some stuff };

executing function will return 
const closureFnReturn = closureFn('some-prop'); // returns (mouseClick) =>  {};

so

```
const routeChange = useCallback((orderId: string) => (event) => history.push('/orders/${orderId}`); 

// then
<Component onClick={routeChange(order.id)} />
```